web/app/js/components/Deployment.jsx

```diff
@@ -188,22 +193,39 @@ export default class Deployment extends React.Component {
     );
   }
 
-  renderContent() {
-    if (_.isEmpty(this.state.metrics)) {
-      return <div>No data</div>;
+  renderDeploymentTitle() {
+    if (!this.state.added){
```

Comment:
tioli we could reduce the repetition here by placing the added check in a ternary:
```
  renderDeploymentTitle() {
    return (<div className="deployment-title">
      <h1>{this.state.deploy}</h1>
      {
        !this.state.added ? (
          <div>
            <div className="status-badge unadded">UNADDED</div>
            <div className={"incomplete-mesh-message"}>
              {this.state.deploy} has not been added to the service mesh
              <div className="instructions">Add the deployment to the deployment.yml file</div>
              <div className="instructions">Then run <code>kubectl inject deployment.yml | kubectl apply -f - </code> to add the deployment to the service mesh</div>
            </div>
          </div>
        ) : null
      }
    </div>);
  }
```